CHANGELOG.md

```diff
@@ -4,6 +4,160 @@ All notable changes to this project will be documented in this file.
+  objects under a field named calls. Former definition syntax is still supported, though deprecated.
+
+  BEFORE:
+
+  ```
+    interceptors: {
+      calls: [
+        (workflowId) => {
+          create(...) => { ... }
+        }
+      ]
+    }
+  ```
+
+  AFTER:
+
+  ```
+    interceptors: [
+      {
+        create(...) => { ... }
+      }
+    ]
+  ```
+
+### Bug Fixes
+
+- [`workflow`] Don't fail workflow task if a query handler was not found ([#932](https://github.com/temporalio/sdk-typescript/pull/932)).
+
+- [`worker`] Wait for worker shutdown if `runUntil` promise throws ([#943](https://github.com/temporalio/sdk-typescript/pull/943)).
+  Previously, `Worker.runUntil` would not wait for worker to complete its shutdown if the inner `fnOrPromise` threw an
+  error. Now, it will always wait for both worker shutdown AND the inner `fnOrPromise` to resolve. If either one throw
+  an error, then that error is rethrown. If _both_ throw an error, a {@link CombinedWorkerRunError} will be thrown
+  instead, with a `cause` attribute containing both errors.
+
+- The (experimental) `FailureConverter` type now receives its `PayloadConverter` through an argument on convertion
+  methods, rather than through an option supplied at construction time ([#936](https://github.com/temporalio/sdk-typescript/pull/936)).
+  This provides a more predictable behaviour in the common case of using the default failure converter. More over,
+  `FailureConverter.errorToFailure` function's return type has been lossen, so that it supports greater customization on
+  user side ([#927](https://github.com/temporalio/sdk-typescript/pull/927))
+
+- [`client`] `ConnectionOptions.connectTimeout` is now being applied correctly ([#954](https://github.com/temporalio/sdk-typescript/pull/954)).
+
+- [`workflow`] Properly encode memos in `makeContinueAsNewFunc` ([#955](https://github.com/temporalio/sdk-typescript/pull/955)).
+  They were previously not encoded at all, resulting in a failure due to invalid data.
+
+- [`worker`] Activity metric `scheduled_to_start_latency` now reports the time from the schedule time of the
+  _current attempt_ to the start time of that same attempt, instead of the time elapsed since the initial schedule time
+  ([#975](https://github.com/temporalio/sdk-typescript/pull/975)). This new definition aligns with other SDKs and is
+  more useful from a monitoring perspective.
+
+- [`workflow`] Previously, `condition(fn, 0)` was incorrectly handled the same as `condition(fn)`, meaning that the
+  function would block indefinitely and would return nothing once `fn` evaluated to true. It now behaves the same as
+  `condition(fn, 1)`, ie. the function will sleep for a very short time, then return true if `fn` evaluates to true,
+  or false if timeout reaches its expiration ([#985](https://github.com/temporalio/sdk-typescript/pull/985)).
+
+- :boom: [`worker`] The experimental API `Worker.runReplayHistories`, used to efficiently replay a large number of
```

Comment:
This was never released so it's technically not a bug fix. The mass replay API should be added to the list of features in the release notes.

I see there are some PRs that didn't use conventional commits that are not included in the generated changelog.
We need to figure out a way to enforce this.